substra/sdk/backends/local/backend.py

```diff
@@ -739,8 +736,309 @@ def describe(self, asset_type, key):
+        for id_, rank in {
+            id_: rank
+            for id_, rank
+            in sorted(visited.items(), key=lambda item: item[1])
+        }.items():
+            if id_ in traintuples:
+                traintuple = traintuples[id_]
+                traintuple_spec = schemas.TraintupleSpec(
+                    algo_key=traintuple.algo_key,
+                    data_manager_key=traintuple.data_manager_key,
+                    train_data_sample_keys=traintuple.train_data_sample_keys,
+                    in_models_keys=[
+                        compute_plan.id_to_key[parent_id] for parent_id in traintuple.in_models_ids
+                    ],
+                    tag=traintuple.tag,
+                    compute_plan_id=compute_plan.compute_plan_id,
+                    rank=rank,
+                    metadata=traintuple.metadata
+                )
+                traintuple = self.add(traintuple_spec, exist_ok, spec_options)
+                compute_plan.id_to_key[id_] = traintuple["key"]
+
+            elif id_ in aggregatetuples:
+                aggregatetuple = aggregatetuples[id_]
+                aggregatetuple_spec = schemas.AggregatetupleSpec(
+                    algo_key=aggregatetuple.algo_key,
+                    worker=aggregatetuple.worker,
+                    in_models_keys=[
+                        compute_plan.id_to_key[parent_id]
+                        for parent_id in aggregatetuple.in_models_ids
+                    ],
+                    tag=aggregatetuple.tag,
+                    compute_plan_id=compute_plan.compute_plan_id,
+                    rank=rank,
+                    metadata=aggregatetuple.metadata
+                )
+                aggregatetuple = self.add(aggregatetuple_spec, exist_ok, spec_options)
+                compute_plan.id_to_key[id_] = aggregatetuple["key"]
+
+            elif id_ in compositetuples:
+                compositetuple = compositetuples[id_]
+                compositetuple_spec = schemas.CompositeTraintupleSpec(
+                    algo_key=compositetuple.algo_key,
+                    data_manager_key=compositetuple.data_manager_key,
+                    train_data_sample_keys=compositetuple.train_data_sample_keys,
+                    in_head_model_key=(compute_plan.id_to_key[compositetuple.in_head_model_id]
+                                       if compositetuple.in_head_model_id else None),
+                    in_trunk_model_key=(compute_plan.id_to_key[compositetuple.in_trunk_model_id]
+                                        if compositetuple.in_trunk_model_id else None),
+                    out_trunk_model_permissions={
+                        "authorized_ids": compositetuple.out_trunk_model_permissions.authorized_ids
+                    },
+                    tag=compositetuple.tag,
+                    compute_plan_id=compute_plan.compute_plan_id,
+                    rank=rank,
+                    metadata=compositetuple.metadata
+                )
```

Comment:
This is just an idea to make the __execute_compute_plan method smaller, not a requirement at all. 

We could have a class method schemas.CompositeTraintupleSpec.from_compute_plan_spec(compute_plan, compute_plan_tuple_spec) that would do all of this (same for the other tuples).

Reply:
That's a good idea, the content of the loop looks like a big blob even though it is only creating the objects.